plugins/user-authenticators/ldap/src/main/java/org/apache/cloudstack/api/command/LinkAccountToLdapCmd.java

```diff
@@ -134,7 +138,14 @@ public String getAdmin() {
     }
 
     public Account.Type getAccountType() {
-        return Account.Type.getFromValue(accountType);
+        if (accountType == null) {
+            return RoleType.getAccountTypeByRole(roleService.findRole(roleId), null);
```

Comment:
@DaanHoogland This will return null if `roleId` is null. Is that fine?
@DaanHoogland Do we need some validations around input as well? e.g. both account type & role id can't be null.

Reply:
I must admit this is not the nicest construct but it is handled in the service: https://github.com/apache/cloudstack/commit/f06e4f22d29c3a2725cab696dbc2e39f6ea8a245